File: Find First Palindromic String in the Array.cpp
class Solution {
public:
    bool ispalindrome(const string& a) {
        for (int i = 0; i < a.size() / 2; i++) {
            if (a[i] != a[a.size() - 1 - i]) {
                return false;
            }
        }
        return true;
    }

    string firstPalindrome(vector<string>& words) {
        for (int i = 0; i < words.size(); i++) {
            if (ispalindrome(words[i])) {
                return words[i];
            }
        }
        return "";
    }
};

File: Leetcode-Problems/Find First Palindromic String in the Array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for (String s : words) {
            if (checkPalindrome(s)) {
                return s;
            }
        }
        return "";
    }

    private boolean checkPalindrome(String s) {
        int i = 0, j = s.length() - 1;
        while (i <= j) {
            if (s.charAt(i) != s.charAt(j))
                return false;
            i++;
            j--;
        }
        return true;
    }
}